examples/fabric/nested_ingestion_pipeline/fabric/tests/test_pl_main_activity.py

```diff
@@ -0,0 +1,102 @@
+import os
+from pathlib import Path
+
+import pytest
+from data_factory_testing_framework import TestFramework, TestFrameworkType
+from data_factory_testing_framework.models import Pipeline
+from data_factory_testing_framework.models.activities import IfConditionActivity
+from data_factory_testing_framework.state import (
+    PipelineRunState,
+)
+
+
+@pytest.fixture
+def test_framework(request: pytest.FixtureRequest) -> TestFramework:
+    return TestFramework(
+        framework_type=TestFrameworkType.Fabric,
+        root_folder_path=os.path.join(Path(request.fspath.dirname).parent, "pl_main.DataPipeline"),
+    )
+
+
+@pytest.fixture
+def pipeline(test_framework: TestFramework) -> Pipeline:
+    return test_framework.get_pipeline_by_name("pl_main")
+
+
+def test_activity_read_configuration_file(request: pytest.FixtureRequest, pipeline: Pipeline) -> None:
+    # Arrange
+    activity = pipeline.get_activity_by_name("Read Configuration File")
+
+    # Act
+    state = PipelineRunState()
+    activity.evaluate(state)
+
+    # Assert
+    expected_data_store_type = "Lakehouse"
+    assert (
+        expected_data_store_type == (activity.type_properties["datasetSettings"]["linkedService"]["properties"]["type"])
+    )
+
+    expected_file_name = "lh_config.json"
+    assert expected_file_name == (activity.type_properties["datasetSettings"]["typeProperties"]["location"]["fileName"])
+
+    expected_folder_name = "config"
+    assert (
+        expected_folder_name
+        == (activity.type_properties["datasetSettings"]["typeProperties"]["location"]["folderPath"])
+    )
+
+
+def test_activity_if(test_framework: TestFramework, pipeline: Pipeline) -> None:
```

Comment:
Should we also introduce a test that tests the negative case? (i.e. result being false)

Reply:
Great idea :) - added!